src/ceph-disk/ceph_disk/main.py

```diff
@@ -303,7 +303,12 @@ class Error(Exception):
 
     def __str__(self):
         doc = _bytes2str(self.__doc__.strip())
-        return ': '.join([doc] + [_bytes2str(a) for a in self.args])
+        try:
+            str_type = basestring
+        except NameError:
+            str_type = str
+        args = [a if isinstance(a, str_type) else str(a) for a in self.args]
```

Comment:
shouldn't this use `str_type` to convert? e.g.:

    args = [a if isinstance(a, str_type) else str_type(a) for a in self.args]